public/app/features/dashboard/timepicker/timepicker.ts

```diff
@@ -96,13 +96,12 @@ export class TimePickerCtrl {
   }
 
   openDropdown() {
+    this.$rootScope.appEvent('escTimepicker');
     if (this.isOpen) {
       this.isOpen = false;
```

Comment:
Here we probably want to emit appEvent that timepicker is closed, i.e. `timepicker-closed`. I think a function closeDropdown would be nice where bot `this.isOpen = false;` and the appEvent is emited. Then you can call this function at the end of `setRelativeFilter` and `applyCustom`.